### PR TITLE
bpo-29469: Optimize literal lists and sets iterating on the AST level.

### PR DESCRIPTION
**Pull request by serhiy-storchaka**
https://bugs.python.org/issue29469

### CODE REVIEW

File: Python/ast_opt.c [hunk 1]
```diff
@@ -268,38 +263,44 @@ fold_subscr(expr_ty node, PyArena *arena)
     return make_const(node, newval, arena);
 }
 
+static int
```
Comment (pitrou) [inline Python/ast_opt.c]: It would be nice to add comments and/or a "docstring" to explain what this is doing.

File: Python/ast_opt.c [hunk 2]
```diff
@@ -268,38 +263,44 @@ fold_subscr(expr_ty node, PyArena *arena)
...
+        }
+    }
+    else {
+        return 1;
+    }
+    return make_const(arg, newval, arena);
+}
+
 static int
 fold_compare(expr_ty node, PyArena *arena)
```
Comment (pitrou) [inline Python/ast_opt.c:291]: Ditto here.

Comment (serhiy-storchaka) [reply Python/ast_opt.c:291]: This already is documented inside the function. In future the optimization for literal operands (like `1 < 2`) can be added here, but currently it optimizes only the particular case of 'in' and 'not in' operators.

**Comment by methane** [comment]
thanks!

**Pull request merged**